lib/pglogical.rb

```diff
@@ -0,0 +1,181 @@
+  def node_drop(name, ifexists = false)
+    typed_exec("SELECT pglogical.drop_node($1, $2)", name, ifexists)
+  end
+
+  # Subscription Management
+  #
+
+  # Creates a subscription to a provider node
+  #
+  # @param name [String] subscription name
+  # @param dsn [String] provider node connection string
+  # @param replication_sets [Array<String>] replication set names to subscribe to
+  # @param sync_structure [Boolean] sync the schema structure when subscribing
+  # @param sync_data [Boolean] sync the data when subscribing
+  # @param forward_origins [Array<String>] names of non-provider nodes to replicate changes from (cascading replication)
+  def subscription_create(name, dsn, replication_sets = %w(default default_insert_only),
+                                    sync_structure = true, sync_data = true, forward_origins = ["all"])
+    command = "SELECT pglogical.create_subscription($1, $2, $3, $4, $5, $6)"
+    typed_exec(command, name, dsn, replication_sets, sync_structure, sync_data, forward_origins)
+  end
+
+  # Disconnects the subscription and removes it
+  #
+  # @param name [String] subscription name
+  # @param ifexists [Boolean] if true an error is not thrown when the subscription does not exist
+  def subscription_drop(name, ifexists = false)
+    typed_exec("SELECT pglogical.drop_subscription($1, $2)", name, ifexists)
+  end
+
+  # Disables a subscription and disconnects it from the provider
+  #
+  # @param name [String] subscription name
+  # @param immediate [Boolean] do not wait for the current transaction before stopping
+  def subscription_disable(name, immediate = false)
+    typed_exec("SELECT pglogical.alter_subscription_disable($1, $2)", name, immediate)
+  end
+
+  # Enables a previously disabled subscription
+  #
+  # @param name [String] subscription name
+  # @param immediate [Boolean] do not wait for the current transaction before starting
+  def subscription_enable(name, immediate = false)
+    typed_exec("SELECT pglogical.alter_subscription_enable($1, $2)", name, immediate)
+  end
+
+  # Syncs all unsynchronized tables in all sets in a single operation.
+  #   Command does not block
+  #
+  # @param name [String] subscription name
+  # @param truncate [Boolean] truncate the tables before syncing
+  def subscription_sync(name, truncate = false)
+    typed_exec("SELECT pglogical.alter_subscription_synchronize($1, $2)", name, truncate)
+  end
+
+  # Resyncs one existing table
+  # Table will be truncated before the sync
+  #
+  # @param name [String] subscription name
+  # @param table [String] name of the table to resync
+  def subscription_resync_table(name, table)
+    typed_exec("SELECT pglogical.alter_subscription_resynchronize_table($1, $2)", name, table)
+  end
+
+  # Adds a replication set to a subscription
+  # Does not sync, only activates event consumption
+  #
+  # @param name [String] subscription name
+  # @param set_name [String] replication set name
+  def subscription_add_rep_set(name, set_name)
```

Comment:
I can handle abbreviating `sync`, but I think `rep_set` (inconsistently, to boot) might be a bridge too far.

(Other than that nit, this is looking awesome :metal:)